tests/components/axis/test_config_flow.py

```diff
@@ -380,6 +382,164 @@ async def test_dhcp_flow_ignore_link_local_address(hass):
+            "ext": "",
+            "server": "Linux/4.14.173-axis8, UPnP/1.0, Portable SDK for UPnP devices/1.8.7",
+            "deviceType": "urn:schemas-upnp-org:device:Basic:1",
+            "friendlyName": f"AXIS M1065-LW - {MAC}",
+            "manufacturer": "AXIS",
+            "manufacturerURL": "http://www.axis.com/",
+            "modelDescription": "AXIS M1065-LW Network Camera",
+            "modelName": "AXIS M1065-LW",
+            "modelNumber": "M1065-LW",
+            "modelURL": "http://www.axis.com/",
+            "serialNumber": MAC,
+            "UDN": f"uuid:Upnp-BasicDevice-1_0-{MAC}",
+            "serviceList": {
+                "service": {
+                    "serviceType": "urn:axis-com:service:BasicService:1",
+                    "serviceId": "urn:axis-com:serviceId:BasicServiceId",
+                    "controlURL": "/upnp/control/BasicServiceId",
+                    "eventSubURL": "/upnp/event/BasicServiceId",
+                    "SCPDURL": "/scpd_basic.xml",
+                }
+            },
+            "presentationURL": f"http://{DEFAULT_HOST}:80/",
+        },
+        context={"source": SOURCE_SSDP},
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == SOURCE_USER
+
+    with respx.mock:
+        mock_default_vapix_requests(respx)
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_HOST: "1.2.3.4",
+                CONF_USERNAME: "user",
+                CONF_PASSWORD: "pass",
+                CONF_PORT: 80,
+            },
+        )
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == f"M1065-LW - {MAC}"
+    assert result["data"] == {
+        CONF_HOST: "1.2.3.4",
+        CONF_USERNAME: "user",
+        CONF_PASSWORD: "pass",
+        CONF_PORT: 80,
+        CONF_MODEL: "M1065-LW",
+        CONF_NAME: "M1065-LW 0",
+    }
+
+    assert result["data"][CONF_NAME] == "M1065-LW 0"
+
+
+async def test_ssdp_flow_already_configured(hass):
+    """Test that SSDP doesn't setup already configured devices."""
+    config_entry = await setup_axis_integration(hass)
+    device = hass.data[AXIS_DOMAIN][config_entry.unique_id]
+    assert device.host == "1.2.3.4"
+
+    result = await hass.config_entries.flow.async_init(
+        AXIS_DOMAIN,
+        data={
+            "friendlyName": f"AXIS M1065-LW - {MAC}",
+            "serialNumber": MAC,
+            "presentationURL": "http://1.2.3.4:80/",
+        },
+        context={"source": SOURCE_SSDP},
+    )
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
+    assert device.host == "1.2.3.4"
+
+
+async def test_ssdp_flow_updated_configuration(hass):
+    """Test that SSDP update configuration with new parameters."""
+    config_entry = await setup_axis_integration(hass)
+    device = hass.data[AXIS_DOMAIN][config_entry.unique_id]
```

Comment:
Please don't access integration details in the test. We don't seem to need the device. We have the config entry available and can check the entry data if needed.

Reply:
I improved the tests https://github.com/home-assistant/core/pull/45489